CMake: Add CMake to platform greentea test (#14072)
ARMmbed/mbed-os: tools/cmake/README.md

```
cmake -S <source-dir> -B <build-dir> -DCMAKE_BUILD_TYPE=debug
```

## How to build a greentea test

Install prerequisites suggested in the previous section and follow the below steps to build:

> * Generate .mbed_build configuration for DISCO_L475VG_IOT01A target from blinky example and copied into the test_suite directory.

Reviewer: ```suggestion
* Generate the `.mbedbuild/` configuration directory for the Mbed target you want to run the test on using [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky)
``'
$ mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> 
``'
* Copy `.mbedbuild/` into the test suite directory.
```

> * Change directory into the test suite directory

Reviewer: ```suggestion
* Set your current directory to the test suite directory
```

> * run below command to build full profile green tea

Reviewer: ```suggestion
* Run the following command to build the test binary with the full profile
```

  ```

>   touch mbed-os.lib;mkdir build;cd build;cmake .. -GNinja;cmake --build .

Reviewer: ```suggestion
  touch mbed-os.lib && mkdir build && cd build && cmake .. -G Ninja && cmake --build .
```

  ```

> * run below command to build baremetal profile green tea

Reviewer: ```suggestion
* Run the following command to build the test binary with the baremetal profile
```

  ```

>   touch mbed-os.lib;mkdir build;cd build;cmake .. -GNinja -DMBED_BAREMETAL_GREENTEA_TEST=ON;cmake --build .

Reviewer: ```suggestion
  touch mbed-os.lib && mkdir build && cd build && cmake .. -G Ninja -DMBED_BAREMETAL_GREENTEA_TEST=ON && cmake --build .
```

```
cmake -S <source-dir> -B <build-dir> -DCMAKE_BUILD_TYPE=debug
```

## How to build a greentea test

Install prerequisites suggested in the previous section and follow the below steps to build:
* Generate the `.mbedbuild/` configuration directory for the Mbed target you want to run the test on using [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky)

> ``'

Reviewer: This needs fixing. I put the wrong stuff on purpose for formatting reasons in my comment. See the closing one too.

$ mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> 
```
* Copy `.mbedbuild/` into the test suite directory.
* Set your current directory to the test suite directory
* Run the following command to build the test binary with the full profile

  ```

>   touch mbed-os.lib && mkdir build && cd build && cmake .. -G Ninja && cmake --build .

Reviewer: ```suggestion
  touch mbed-os.lib && mkdir cmake_build && cd cmake_build && cmake .. -G Ninja && cmake --build .
```
To be in line with what `mbedtools` creates.

Reviewer 2: @rajkan01 Can you make the requested edits ?

  ```
* Run the following command to build the test binary with the baremetal profile
  ```

>   touch mbed-os.lib && mkdir build && cd build && cmake .. -G Ninja -DMBED_BAREMETAL_GREENTEA_TEST=ON && cmake --build .

Reviewer: ```suggestion
  touch mbed-os.lib && mkdir cmake_build && cd cmake_build && cmake .. -G Ninja -DMBED_BAREMETAL_GREENTEA_TEST=ON && cmake --build .
```
To be in line with what `mbedtools` creates.

  ```

> Note: This steps will get evolve once mbedtools have a proper way of invoking greentea test using mbedtools command 

Reviewer: ```suggestion
Note: These steps will change when `mbedtools` implements a sub-command to invoke Greentea tests
```
